src/server/src/main/java/io/cassandrareaper/service/RepairRunner.java

```diff
@@ -272,10 +282,108 @@ private void endRepairRun() {
 
         context.metricRegistry.counter(
           MetricRegistry.name(RepairManager.class, "repairDone", RepairRun.RunState.DONE.toString())).inc();
+
+        maybeAdaptRepairSchedule();
       }
     }
   }
 
+  /**
+   * Tune segment timeout and number of segments for adaptive schedules.
+   * Checks that the run was triggered by an adaptive schedule and gathers info on the run to apply tunings.
+   */
+  @VisibleForTesting
+  public void maybeAdaptRepairSchedule() {
```

Comment:
I don't see the need for `@VisibleForTesting` or `public` 🤷🏻‍♀️

Reply:
Good catch, it's a leftover from trials on unit testing. I'll move it back to private.

Reply:
scratch that, I actually need this method to be `public` so that I can invoke it from the RepairRunner spy I'm creating ([here](https://github.com/thelastpickle/cassandra-reaper/pull/1110/files#diff-e06412fc9bcb60ad7dfbbce3c7cf2b725b2fe853e3fe95fe8461abd3b64412e8R1239) for example).

I'll revert the change.

Reply:
Can spy work with package-protected?  (which would be better than public)

Reply:
I just tried and it doesn't work, sorry.